Content.Shared/Magic/Systems/AnimateSpellSystem.cs

```diff
@@ -0,0 +1,55 @@
+using Content.Shared.Item;
+using Content.Shared.Magic.Components;
+using Content.Shared.Physics;
+using Robust.Shared.Containers;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Physics.Systems;
+using System.Linq;
+
+namespace Content.Shared.Magic.Systems;
+
+public sealed class AnimateSpellSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<AnimateComponent, ComponentStartup>(OnAnimate);
+        SubscribeLocalEvent<AnimateComponent, GettingPickedUpAttemptEvent>(OnPickupAttempt);
+    }
+
+    private void OnAnimate(EntityUid uid, AnimateComponent comp, ComponentStartup args)
+    {
+        // Physics bullshittery necessary for object to behave properly
+
```

Comment:
```suggestion

    private void OnAnimate(Entity<AnimateComponent> ent, ref OnMapInit args)
    {
        // Physics bullshittery necessary for object to behave properly

```

We use Entity<T> now instead. Code below should be updated to use ent.Owner instead of uid and ent.Comp instead of comp